### PR TITLE
feat: run system commands

### PR DESCRIPTION
**Pull request by BugenZhao**
Support running system commands in `slt` files.

```
system ok
echo "hello, world" > __TEST_DIR__/a.txt
```

Signed-off-by: Bugen Zhao <i@bugenzhao.com>

### CODE REVIEW

File: sqllogictest-bin/src/engines.rs [hunk 1]
```diff
@@ -113,6 +115,22 @@ impl AsyncDB for Engines {
...
+        dispatch_engines!(self, e, { e.engine_name() })
+    }
+
+    async fn sleep(dur: Duration) {
+        tokio::time::sleep(dur).await
+    }
+
+    async fn run_command(command: std::process::Command) -> std::io::Result<ExitStatus> {
+        Command::from(command).status().await
     }
```
Comment (liurenjie1024) [inline sqllogictest-bin/src/engines.rs:135]: This is a breaking api change. Is it possible to make them default methods?

**Review by liurenjie1024 — APPROVED** [review]

**Review by wangrunji0408 — APPROVED** [review]
LGTM!

File: sqllogictest-bin/src/engines.rs (hunk 1 above)
Comment (BugenZhao) [reply sqllogictest-bin/src/engines.rs:135]: It does have default implementations. However, new variants in `RecordOutput` and `TestErrorKind` still break the API. I guess we have to bump the major version.

**Comment by BugenZhao** [comment]
Bump the version to v0.16.0 and also include the change log from #194.

**Review by xxchan — APPROVED** [review]

**Pull request merged**